README.md

```diff
@@ -346,21 +315,116 @@ const config = {
 context. In this repository we use the `.env` file for storing sandbox data.
 
-## Migration guide from the old checkout JS
+## Migration guide
+
+1. Look for the following scripts:
+
+```html
+<script src="https://code.jquery.com/jquery-1.12.4.min.js"></script>
+<script src="https://checkout.freemius.com/checkout.min.js"></script>
+```
+
+2. Remove the jQuery script tag if you aren't using jQuery.
+3. Replace the checkout script with the new one.
 
-If you've been using the old `checkout.freemius.com/checkout.min.js` script then
-this guide is for you. We have introduced a compatibility layer using which you
-can very easily migrate to the new checkout JS.
+```html
+<script src="https://checkout.freemius.com/js/v1/"></script>
+```
+
+4. Change `FS.Checkout.configure()` to `new FS.Checkout()`:
+
+```js
+// Legacy checkout code
+const handler = FS.Checkout.configure({
+    plugin_id: '1234',
+    plan_id: '5678',
+    public_key: 'pk_ccca7be7fa43aec791448b43c6266',
+    image: 'https://your-plugin-site.com/logo-100x100.png',
+});
+```
+
+```js
+// New checkout code
+const handler = new FS.Checkout({
+    plugin_id: '1234',
+    plan_id: '5678',
+    public_key: 'pk_ccca7be7fa43aec791448b43c6266',
+    image: 'https://your-plugin-site.com/logo-100x100.png',
+});
+```
+
+The rest of the code will continue to work exactly as it is with no changes.
+
+```js
+document.querySelector('#purchase').addEventListener('click', (e) => {
+    handler.open({
+        name: 'My Awesome Plugin',
+        licenses: document.querySelector('#licenses').value,
+        // You can consume the response for after purchase logic.
+        purchaseCompleted: function (response) {
+            // The logic here will be executed immediately after the purchase confirmation.
+            // alert(response.user.email);
+        },
+        success: function (response) {
+            // The logic here will be executed after the customer closes the checkout, after a successful purchase.
+            // alert(response.user.email);
+        },
+    });
 
-In your code, where you do
+    e.preventDefault();
+});
+```
+Note: If you need to add a checkout for a different configuration on the same page, just create a new checkout:
+
+```js
+const anotherHandler = new FS.Checkout({
+    plugin_id: '4321',
+    plan_id: '9876',
+    public_key: 'pk_....nnn',
+    image: 'https://your-plugin-site.com/logo-100x100.png',
+});
+```
+
+Now you can add another event listener that opens the new checkout:
+
+```js
+document.querySelector('#another-purchase-button').addEventListener('click', (e) => {
+    anotherHandler.open({
+        name: 'My Awesome Plugin',
+        licenses: document.querySelector('#licenses').value,
+        purchaseCompleted: function (response) {
+            //...
+         },
+        success: function (response) {
+            //...
+        },
+    });
+
+    e.preventDefault();
+});
+```
+
```

Comment:
Let's also add another tip for migrating the legacy singleton. Something like:

> If you've been using the `FS.Checkout` singleton interface like
> ```js
> FS.Checkout.open({ 
>     plugin_id: 'x', 
>     // ...
> });
> ```
> Then you will need to adjust your code to call the methods on the instance instead of the singleton.
> ```js
> //... Add another example
> ```